src/iter/fold.rs

```diff
@@ -114,4 +118,8 @@ impl<'r, C, ID, F, T> Folder<T> for FoldFolder<'r, C, ID, F>
     fn complete(self) -> C::Result {
         self.base.consume(self.item).complete()
     }
+
+    fn full(&self) -> bool {
+        self.base.full()
+    }
```

Comment:
Should we make `full` required instead of optional, to prevent us from forgetting it?